### DISCUSSION TITLE
Train the Yolov8 Segmentation Model with simply mask pictures

### DISCUSSION THREAD

**Discussion by Hydrageno** [ultralytics/ultralytics#17134] [Q&A] [answered]
Hello Everyone !!
I have a dataset with **only mask pictures**. I wannted to train the **Yolov8 Segmentation Model** with these masks. 
And There is my **training .yaml:**
```
train: /home/test/SegmentationTest/images/train
val: /home/test/SegmentationTest/images/val
test: /home/test/SegmentationTest/images/test
# Classes
nc: 9
names: ['Background', 'Car', 'People', 'Bike', 'Curve', 'Sign', 'Fencing', 'Barricade', 'GroundRaised']
colors: [[0,0,0], [64,0,128], [64,64,0], [0,128,192], [0,0,192], [128,128,0], [64,64,128], [192,128,128], [192,64,0]]
```
my folder structure be like:
```
SegmentationTest/
├── images/
│   ├── train/
│   │   ├── image1.jpg
│   │   ├── image2.jpg
│   │   └── ...
│   └── val/
│       ├── image1.jpg
│       ├── image2.jpg
│       └── ...
└── masks/
    ├── train/
    │   ├── image1.png
    │   ├── image2.png
    │   └── ...
    └── val/
        ├── image1.png
        ├── image2.png
        └── ...
```
**However**:
```
FileNotFoundError: train: No labels found in /home/test/SegmentationTest/labels/train.cache
```
during the training process, 
Looking for the solutions, thank you so much!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Hydrageno, thank you for reaching out and showing interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com) for new users where you can explore detailed [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples to guide you through common applications.

If this is a 🐛 Bug Report, it seems like you are encountering a `FileNotFoundError`. To help us assist you better, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). This will assist in debugging your issue more efficiently.

If this is a custom training ❓ question, please ensure to include all the relevant information, such as sample images from your dataset and detailed training logs. Moreover, it's beneficial to review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

In the meantime, if you want to join discussions or need real-time help, consider joining our [Discord community](https://ultralytics.com/discord) 🎧. Alternatively, visit our [Discourse platform](https://community.ultralytics.com) for comprehensive discussions, or join our [Subreddit](https://reddit.com/r/ultralytics) to share insights.

## Upgrade

Make sure you are using the latest version of the `ultralytics` package along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) within a [**Python>=3.8**](https://www.python.org/) environment and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). This helps ensure that any potential bugs you are experiencing are not already fixed in the latest version:

```bash
pip install -U ultralytics
```

## Environments

Run YOLO in verified environments that are pre-configured with all essential dependencies like [CUDA](https://developer.nvidia.com/cuda), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. Follow the instructions in the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. Refer to the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. Use the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

Rest assured, an Ultralytics engineer will also assist you soon. If your status badge is green, it confirms that all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. CI tests check the accurate operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu every 24 hours, and on each commit.

**Reply by Hydrageno** [reply]
I have check the environment. And there is my **training.py**
```
from ultralytics import YOLO
model = YOLO('pretrained/yolov8n-seg.pt') 
if __name__ == '__main__':
    results = model.train(data='data/training.yaml', epochs=10, imgsz=512)
    metrics = model.val()
```

**Reply by glenn-jocher** [reply]
@Hydrageno your script looks good for training. Ensure your dataset paths and annotations are correctly set up in `training.yaml`. If you encounter issues, please share the specific error message for further assistance.

**Reply by Hydrageno** [reply]
**Thank you**, I have solved the problem by changing the **mask png** to **yolo format**. 
And there is **another question**. 
The corresponding text does not include the "Background", should I remove the class in the data.yml just like:
```
names: ['Car', 'People', 'Bike', 'Curve', 'Sign', 'Fencing', 'Barricade', 'Bump']
```
Or remain:
```
names: ['BackGround', 'Car', 'People', 'Bike', 'Curve', 'Sign', 'Fencing', 'Barricade', 'Bump']
```

**Reply by glenn-jocher** [reply, marked as answer]
If "Background" is not annotated, you should remove it from the `names` list in your `data.yaml`.

**Reply by Hydrageno** [reply]
Thank you so much!!!

**Reply by glenn-jocher** [reply]
@Hydrageno you're welcome! If you have any more questions, feel free to ask.

**Comment by glenn-jocher** [comment]
@Hydrageno based on your folder structure and error message, I notice you're missing the required "labels" directory for YOLO segmentation training. For segmentation tasks, you need to convert your mask images into YOLO format labels. Here's what you need to do:

1. Create a "labels" directory parallel to your "images" directory
2. Convert your PNG masks into YOLO segmentation format (txt files) and place them in the labels directory
3. Update your dataset structure to match:

```
SegmentationTest/
├── images/
│   ├── train/
│   └── val/
└── labels/
    ├── train/
    └── val/
```

The YOLO format requires polygon coordinates in normalized form (0-1) for each instance in the mask. You can use tools like labelme or our dataset conversion utilities to convert your mask images to the required YOLO format.

I recommend checking our segmentation guide at docs.ultralytics.com/tasks/segment/ for detailed instructions on proper dataset formatting and training procedures.
